tests/individual/TimeTravel.cpp

```diff
@@ -0,0 +1,145 @@
+
+class TestReactor : public test_util::TestBase<TestReactor, 5000> {
+public:
+    // Start time of steady clock
+    std::chrono::steady_clock::time_point steady_start_time = std::chrono::steady_clock::now();
+
+    // Time travel action
+    NUClear::message::TimeTravel::Action action = NUClear::message::TimeTravel::Action::RELATIVE;
+
+    // Time adjustment
+    NUClear::clock::duration adjustment = std::chrono::milliseconds(0);
+
+    // Real-time factor
+    double rtf = 1.0;
+
+    // Results struct
+    Results results;
+
+    TestReactor(std::unique_ptr<NUClear::Environment> environment) : TestBase(std::move(environment), false) {
+
+        on<Startup>().then([this] {
+            results.zero = Results::TimePair{NUClear::clock::now(), std::chrono::steady_clock::now()};
+
+            // Emit a chrono task to run at time EVENT_1_TIME
+            emit<Scope::DIRECT>(std::make_unique<NUClear::dsl::operation::ChronoTask>(
+                [this](NUClear::clock::time_point&) {
+                    results.events[0] = Results::TimePair{NUClear::clock::now(), std::chrono::steady_clock::now()};
+                    return false;
+                },
+                NUClear::clock::now() + TestUnits(EVENT_1_TIME),
+                1));
+
+            // Emit a chrono task to run at time EVENT_2_TIME, and shutdown
+            emit<Scope::DIRECT>(std::make_unique<NUClear::dsl::operation::ChronoTask>(
+                [this](NUClear::clock::time_point&) {
+                    results.events[1] = Results::TimePair{NUClear::clock::now(), std::chrono::steady_clock::now()};
+                    powerplant.shutdown();
+                    return false;
+                },
+                NUClear::clock::now() + TestUnits(EVENT_2_TIME),
+                2));
+
+            // Time travel!
+            emit<Scope::DIRECT>(
+                std::make_unique<NUClear::message::TimeTravel>(NUClear::clock::time_point(adjustment), rtf, action));
+
+            results.start = Results::TimePair{NUClear::clock::now(), std::chrono::steady_clock::now()};
+        });
+    }
+};
+
+}  // anonymous namespace
+
+TEST_CASE("Test time travel correctly changes the time for non zero rtf", "[time_travel][chrono_controller]") {
+
+    using Action = NUClear::message::TimeTravel::Action;
+
+    const NUClear::Configuration config;
+    auto plant    = std::make_shared<NUClear::PowerPlant>(config);
+    auto& reactor = plant->install<TestReactor>();
+
+    // Set the reactor fields to the values we want to test
+    const Action action      = GENERATE(Action::RELATIVE, Action::ABSOLUTE, Action::NEAREST);
+    const int64_t adjustment = GENERATE(-4, -2, 0, 2, 4, 6, 8, 10);
+    const double rtf         = GENERATE(0.5, 1.0, 2.0);
+    CAPTURE(action, adjustment, rtf);
+    reactor.action     = action;
+    reactor.adjustment = TestUnits(adjustment);
+    reactor.rtf        = rtf;
+
+    // Reset clock to zero
+    NUClear::clock::set_clock(NUClear::clock::time_point());
```

Comment:
Maybe we should move this into the on<Startup> as a direct emit right before the zero to make sure we don't encounter any clock jitter